nixos/modules/services/mail/listmonk.nix

```diff
@@ -0,0 +1,212 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.listmonk;
+  tomlFormat = pkgs.formats.toml {};
+  cfgFile = tomlFormat.generate "listmonk.toml" cfg.settings;
+  setDatabaseOption = key: value: "UPDATE settings SET value = '${builtins.toJSON value}' WHERE key = '${key}';";
+  updateDatabaseConfigSQL = pkgs.writeText "update-database-config.sql"
+    (concatStringsSep "\n" (mapAttrsToList setDatabaseOption (if (cfg.databaseSettings != null) then cfg.databaseSettings else {})));
+  updateDatabaseConfigScript = pkgs.writeShellScriptBin "update-database-config.sh" ''
+    ${if cfg.mutableDatabaseSettings then ''
+      if [ ! -f $STATE_DIRECTORY/.db_settings_initialized ]; then
+        ${pkgs.postgresql}/bin/psql -d listmonk -f ${updateDatabaseConfigSQL} ;
+        touch $STATE_DIRECTORY/.db_settings_initialized
+      fi
+    '' else "${pkgs.postgresql}/bin/psql -d listmonk -f ${updateDatabaseConfigSQL}"
+    }
+  '';
+in
+{
+  ###### interface
+  options = {
+    services.listmonk = {
+      enable = mkEnableOption "Listmonk";
+      database.createLocally = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc "Create the PostgreSQL database and database user locally.";
+      };
+      package = mkOption {
+        type = types.package;
+        default = pkgs.listmonk;
+        defaultText = lib.literalExpression "pkgs.listmonk";
+        description = lib.mdDoc "Package to use for this service";
+      };
+      settings = mkOption {
+        type = types.submodule {
+          freeformType = tomlFormat.type;
+        };
+        description = lib.mdDoc ''
+          Static settings set in the config.toml, see <https://github.com/knadh/listmonk/blob/master/config.toml.sample> for details.
+          You can set secrets using the secretFile option with environment variables following <https://listmonk.app/docs/configuration/#environment-variables>.
+        '';
+      };
+      mutableDatabaseSettings = mkOption {
+        type = types.bool;
+        default = true;
+        description = lib.mdDoc ''
+          Database settings will be reset to the value set in this module if this is not enabled.
+          Enable this if you want to persist changes you have done in the application.
+        '';
+      };
+      databaseSettings = mkOption {
```

Comment:
A nitpick but it is inconsistent that you have `database.createLocally` and then `databaseSettings`. Maybe rename to `database.settings` if you agree.

Reply:
Addressed.